### PR TITLE
Screenshots collector + extract reporters + HTML reporter

### PR DESCRIPTION
**Pull request by kdzwinel**
Example call:

```
npm run crawl -- -i ./data/10.txt -o ./data/10/ -d 'requests,screenshots' --reporters 'html' -l './data/' -f -v
```

- all reporters (CLI, file, HTML) are extracted to separate folder (`/reporters/`)
- images in HTML file are lazyloaded (thanks to https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/loading)
- screenshots are now a proper collector
- screenshots don't have to be moved, they are saved next to JSON data files right away, so no need for `fs-extra` package

Things that changed:
- screenshots next to data, not in a separate folder
- one thing that's missing is that you can't specify for which urls screenshot should be taken - we have to do two crawls (one with `-d 'screenshots'` (for those domains where you want to make a screenshot) and one without (for those domains where you don't want to make a screenshot)
- `-l` was used to set path to TXT log, now it's used to set path for all logs (both TXT and HTML)

### CODE REVIEW

File: cli/crawl-cli.js [hunk 1]
```diff
@@ -95,35 +100,49 @@ async function run(inputUrls, outputPath, verbose, logPath, numberOfCrawlers, da
...
      */
     const dataCallback = (url, data) => {
         successes++;
-        updateProgress(url.toString());
+
+        crawlTimes.push([data.testStarted, data.testFinished, data.testFinished - data.testStarted]);
 
         const outputFile = createOutputPath(url);
+
+        // temp name for the screenshot is scored in data. rename the screenshot to match the file crawl file
```
Comment (sammacbeth) [inline cli/crawl-cli.js:135]: I think this comment is now out of date (screenshot is not renamed)

**Review by sammacbeth — APPROVED** [review]
This is great! I already have some ideas of additional reporters we could write :)

File: cli/crawl-cli.js (hunk 1 above)
Comment (kdzwinel) [reply cli/crawl-cli.js:135]: ah, good call, I'll fix that. Thanks for reviewing!

**Comment by jdorweiler** [comment]
Thanks for doing this!

> screenshots don't have to be moved, they are saved next to JSON data files right away

Can we have an option for separate directory path to put the files in?

> can't specify for which urls screenshot should be taken

Is this difficult to add?

**Review by jdorweiler — APPROVED** [review]
🚀

**Pull request merged**